### PR TITLE
chore(argo-workflows): Upgrade argo-workflows to 3.5.0

### PR DESCRIPTION
**Pull request by jmeridth**
https://github.com/argoproj/argo-workflows/releases/tag/v3.5.0

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [ ] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [ ] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

**Comment by jmeridth** [comment]
Will fail tests until the official image is released after upstream 3.5.0 release happens.

**Comment by csantanapr** [comment]
Hi @jmeridth what's the process to bump a chart version? 
Do you compare the previous release manifest yamls to the new release, and inspect what needs to be updated for the new chart?
I want to understand the process to see if I can help and contribute on this repo

**Comment by jmeridth** [comment]
@csantanapr welcome.  all help is appreciated.

Yes, we inspect the diffs and update manifests and update our templates/values as needed.

I need to update this PR due to rc1 and rc2 releases.  Will do that today.

**Comment by csantanapr** [comment]
Hi @jmeridth Is this something you do manually, or you have some helper scripts to find the diffs?

**Comment by jmeridth** [comment]
@csantanapr manual currently.

I look at the diffs upstream via https://github.com/argoproj/argo-workflows/compare/v3.4.11...v3.5.0-rc2 currently and look specifically at any CRD changes in the `manifests/base/crds/minimal` folder

**Comment by jmeridth** [comment]
Tests are failing due to Quay.io outage

https://status.quay.io/incidents/3dq7w1tvs16f Will watch this.

**Review by tico24 — APPROVED** [review]

**Review by mbevc1 — APPROVED** [review]

File: charts/argo-workflows/Chart.yaml [hunk 1]
```diff
@@ -17,4 +17,4 @@ annotations:
     url: https://argoproj.github.io/argo-helm/pgp_keys.asc
   artifacthub.io/changes: |
     - kind: changed
-      description: Configure declarative server auth mode.
+      description: Upgrade to Argo Workflows v3.4.10
```
[the remark this answers is not in the document]
Comment (jmeridth) [reply charts/argo-workflows/Chart.yaml:20]: @RoryDoherty good catch. I will look into this.

**Pull request merged**